core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -124,18 +127,30 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
                                       networkClient: NetworkClient,
                                       txnStateManager: TransactionStateManager,
                                       txnMarkerPurgatory: DelayedOperationPurgatory[DelayedTxnMarker],
-                                      time: Time) extends Logging {
+                                      time: Time,
+                                      metrics:Metrics) extends Logging {
+
+  private def addQueueLengthMetric(sensor: Sensor, namePrefix: String): Unit = {
+    sensor.add(metrics.metricName(s"$namePrefix-queue-length", "transaction-marker-channel-metrics"), new Total())
```

Comment:
The reason for doing something different is because the `Metrics` class was there and I'd assumed that was the way it was done...